### PR TITLE
Remove `ContactMap` (allowing rename `ContactFrequency` => `ContactMap`)

### PR DESCRIPTION
**Pull request by dwhswenson**
This is the next step in #82. Removes `ContactMap` and adds `PendingDeprecationWarning` on `ContactFrequency`.

This also bumps the version to 0.7.0.dev0 (which should have been done as soon as we broke the API in #87; changed API means we're developing on the next 0.x).

### CODE REVIEW

File: contact_map/contact_map.py [hunk 1]
```diff
@@ -686,9 +637,15 @@ class ContactFrequency(ContactObject):
     """
     # Default for use_atom_slice, None tries to be smart
     _class_use_atom_slice = None
+    _pending_dep_msg = (
+        "ContactFrequency will be renamed to ContactMap in version 0.8. "
+        + "Invoking it as ContactFrequency will be deprecated in 0.8. For "
+        + "more, see https://github.com/dwhswenson/contact_map/issues/82"
```
Comment (sroet) [inline contact_map/contact_map.py:643]: plusses are not needed inside brackets
```suggestion
        "Invoking it as ContactFrequency will be deprecated in 0.8. For "
        "more, see https://github.com/dwhswenson/contact_map/issues/82"
```
Lead to better indentation (in my opinion)

**Review by sroet — CHANGES_REQUESTED** [review]
Only a comment about combining strings without `+` inside brackets, for cleaner indentation.

LGTM otherwise

**Comment by dwhswenson** [comment]
Desired changes are done ... I had completely forgotten that this was still sitting here!

**Review by sroet — APPROVED** [review]
LGTM, feel free to merge

**Comment by dwhswenson** [comment]
Update #82 to reflect current status.

**Pull request merged**